### ISSUE TITLE
Optional Washer Door Lock

### ISSUE THREAD

**Issue by jstoff** [nVuln/homebridge-lg-thinq#69]
Hi! I've love it if I could hide the Washer Door Lock item in HomeBridge — maybe via a checkbox in the UI. I would rather not see it because, by default, it sends notifications whenever the washer door locks or unlocks.

`[02/09/2021, 13:06:41] [LGThinQ] Found device:  {"deviceId":"75e0a19e-8865-116e-a3c5-7440beee3f5e","deviceType":201,"modelName":"T1789EFH_F","subModelNm":null,"sensorType":null,"alias":"Washer","deviceCode":"LA01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"TYPE2","pccModelYn":"Y","autoOrderYn":"N","drServiceYn":"N","ssid":"StoffPlume","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"LA01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20200619020528000","regIndex":0,"blackboxYn":"Y","groupableYn":"N","controllableYn":"N","combinedProductYn":"N","masterYn":"Y","snapshot":{"washerDryer":{"initialBit":"INITIAL_BIT_OFF","initialTimeMinute":26,"preState":"END","dnn_temp":"17","addGarment":"ADDGARMENT_OFF","error":"ERROR_NO","deepFillMax":"DEEPFILLMAX_OFF","dnn_precipitationProbability":"0","dnn_washingIndex":"80","dnnReady":"DNNREADY_OFF","remainTimeHour":0,"preWash":"PREWASH_OFF","steam":"STEAM_OFF","state":"POWEROFF","rinse":"NO_RINSE","courseTLBaseUS":"NOT_SELECTED","smallUE":"SMALLUE_OFF","temp":"NO_TEMP","doorLock":"DOOR_LOCK_OFF","reserveTimeMinute":0,"smartCourseTLBaseUS":"NOT_SELECTED","washingIndex":"80","dnn_humidity":"38","TCLCount":20,"stainCare":"STAINCARE_OFF","waterPlus":"WATERPLUS_OFF","fabricSoftener":"FABRICSOFTENER_OFF","saveEnergy":"SAVE_ENERGY_OFF","wifiSDS":"WIFI_SDS_OFF","extraRinse":"EXTRARINSE_OFF","dnn_dust":"-","turboWash":"TURBOWASH_ON","remainTimeMinute":0,"reserveTimeHour":0,"soak":"SOAK_OFF","childLock":"CHILDLOCK_OFF","remoteStart":"REMOTE_START_OFF","spin":"NO_SPIN","soakBit":"SOAK_OFF","downloadedCourseTLBaseUS":"SMALL_LOAD","soilWash":"NO_SOILWASH","coldWash":"COLDWASH_OFF","deepFill":"DEEPFILL_OFF","initialTimeHour":0},"fwStatus":"complete","static":{"deviceType":"201","countryCode":"US"},"fwCurrentCount":0,"meta":{"allDeviceInfoUpdate":true,"messageId":"aZWVuVpLSeeyYcdoa9Ulgg"},"mid":443546605,"online":true,"fwCurrentFileSize":0,"fwCount":1,"timestamp":1630602902894},"manufacture":null,"online":true,"platformType":"thinq2","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"N","modelJsonVer":"3.5","modelJsonUri":"https://objectcontent.lgthinq.com/f42b2a5d-7d0e-47a0-84a7-7a561702e7e3?hdnts=exp=1666904843~hmac=5d012072d7c106f89e5ab57e824a4b3dd36e921e55fbdb22a1192898173df260","appModuleVer":"3.63","appModuleUri":"https://objectcontent.lgthinq.com/27342417-7989-4cd5-a248-9b204c437c3d?hdnts=exp=1690080696~hmac=57e58b829363251d0615a494886d685f1148c1f74649cb04d9fd7a19ebd3484b","appRestartYn":"Y","appModuleSize":"9896643","langPackProductTypeVer":"83.2","langPackProductTypeUri":"https://objectcontent.lgthinq.com/9471ee32-b8b3-47d7-b631-3076dc0078c2?hdnts=exp=1693121763~hmac=239a1962c07fd315498cdbd61ba718370ef54b6641027ffe46addd999ca56f0e","langPackModelVer":"1.3","langPackModelUri":"https://objectcontent.lgthinq.com/753dcaf1-5470-42aa-9ff3-1983ec9be6cc?hdnts=exp=1681869628~hmac=8c6b2a0f421cb035fe22706d6ef6f68c9e5048cc8a1ee6a0658ef51d5938fe27","roomId":"160079885009562145","fwInfoList":[{"checksum":"0000C250","partNumber":"SAA39955103","order":2},{"checksum":"00003A43","partNumber":"SAA39954903","order":1}],"modemInfo":{"modelName":"T1789EFH_F","appVersion":"clip_hna_v1.9.104","modemType":"QCOM_QCA4010","ruleEngine":"y"},"existsEntryPopup":"N","fwVer":null,"modemVer":null,"subDeviceCount":0,"firebaseLogKey":"T:TL-S:27-U:5-P:TD","cardType":"Small","cardControl":"Base"}
[02/09/2021, 13:06:41] [LGThinQ] Restoring existing accessory from cache: Washer`

**Comment by nVuln** [comment]
You can quickly hide it and turn off notification in home app
![image](https://user-images.githubusercontent.com/54855446/131942864-925f1914-7431-4338-9a11-825017e5d367.jpeg)

**Comment by jstoff** [comment]
I know - but it's a pain to configure each family member's notifications/status, because HomeKit doesn't let us set defaults for an accessory. So toggling the "door lock" at the homebridge level would be preferable.

**Comment by tech23y** [comment]
Indeed would be great if the door could be excluded at hb ui  because when you ask siri for washer status it always tells you the door status :)

**Comment by nVuln** [comment]
> Indeed would be great if the door could be excluded at hb ui because when you ask siri for washer status it always tells you the door status :)

understood, I'll update soon

**Comment by nVuln** [comment]
I already added this feature in latest version, go to plugin setting and turn it off, make sure `washer_door_lock = false` in config

**Comment by falinka** [comment]
Sorry to 'hijack' this thread, but maybe the same thing could be applied to dryers as well?

**Comment by nVuln** [comment]
> Sorry to 'hijack' this thread, but maybe the same thing could be applied to dryers as well?

at this time, it's already applied to Dryer too, put `washer_door_lock = false` to device config seem as washer and it will work
I'll update config schema in next release

**Comment by falinka** [comment]
I did disable it via UI, checked the config file ("washer_door_lock": false), restarted HB, but somehow, lock is still present in HomeKit. Wondering if I'm doing something wrong though.

**Comment by nVuln** [comment]
> I did disable it via UI, checked the config file ("washer_door_lock": false), restarted HB, but somehow, lock is still present in HomeKit. Wondering if I'm doing something wrong though.

can you upload your config ?
try clean homebridge accessory cache also, maybe door lock get cached

**Comment by falinka** [comment]
I did clear all the cached accessories, restarted, washer door lock status is still present, but the dryer lock is gone 😕 

I could drop debug log as well, if it helps..

```
        {
            "username": "xxxx",
            "password": "xxxx",
            "country": "CZ",
            "language": "en-CZ",
            "thinq1": false,
            "washer_trigger": false,
            "platform": "LGThinQ",
            "auth_mode": "token",
            "refresh_token": "xxxx",
            "devices": [
                {
                    "id": "xxx",
                    "name": "Washer",
                    "type": "WASHER",
                    "washer_door_lock": false
                },
                {
                    "id": "xxx",
                    "name": "Dryer",
                    "type": "DRYER"
                },
                {
                    "id": "xxx",
                    "name": "Refrigerator",
                    "type": "REFRIGERATOR"
                }
            ],
            "refresh_interval": 60
        }
```

**Comment by nVuln** [comment]
> I did clear all the cached accessories, restarted, washer door lock status is still present, but the dryer lock is gone 😕

is the lock still present in Homebridge Accessories tab ?

**Comment by falinka** [comment]
Yes, it is, for washer.

**Comment by nVuln** [comment]
> Yes, it is, for washer.

update to `v1.2.3` and try again, hope it's work

**Comment by falinka** [comment]
> > Yes, it is, for washer.
> 
> update to `v1.2.3` and try again, hope it's work

Yeap, finally, it does. It took some on/off sequence, event though it's by default disabled, initially, was shown. Then I toggled "display", restart, set value in config to "false", restart.. et voila.

Also, what I've noticed, is that the value in config file was missing, i.e. there was no 'washer_door_lock' at all. Only once you enable/disable, it actually pops up in config file, with corresponding value, true/false.

Thank you so much, beer is on me.
